src/react-hooks/buildHooks.ts

```diff
@@ -229,6 +239,78 @@ export function buildHooks<Definitions extends EndpointDefinitions>({
       );
     };
 
+    const useLazyQuerySubscription: UseLazyQuerySubscription<any> = ({
+      refetchOnReconnect,
+      refetchOnFocus,
+      pollingInterval = 0,
+    } = {}) => {
+      const { initiate } = api.endpoints[name] as ApiEndpointQuery<
+        QueryDefinition<any, any, any, any, any>,
+        Definitions
+      >;
+      const dispatch = useDispatch<ThunkDispatch<any, any, AnyAction>>();
+
+      const promiseRef = useRef<QueryActionCreatorResult<any>>();
+      const lastPromise = promiseRef.current;
+
+      const optionsRef = useRef<SubscriptionOptions>();
+      const argsRef = useRef<any>();
+
+      useEffect(() => {
+        const options = {
+          refetchOnReconnect,
+          refetchOnFocus,
+          pollingInterval,
+        };
+        if (lastPromise) {
+          if (!shallowEqual(options, optionsRef.current)) {
+            lastPromise.updateSubscriptionOptions(options);
+            optionsRef.current = options;
+          }
+        }
```

Comment:
```suggestion
        if (optionsRef.current && !shallowEqual(options, optionsRef.current)) {
          lastPromise.updateSubscriptionOptions(options);
          optionsRef.current = options;
        }
```
maybe just tie it to `optionsRef` already being set so it's not two different values kinda tracking the same thing?

Reply:
We could also think about moving a "subscriptionOptions" value into the promise over in `buildInitiate` as having access to `promiseRef.current.subscriptionOptions` as the last updated subscriptionOptions might be valuable outside of this hook in other implementations as well?

Reply:
Updated and used as the check :+1: